pymbolic/algorithm.py

```diff
@@ -137,13 +141,45 @@ def fft(x, sign=1, wrap_intermediate=lambda x: x, complex_dtype=None):
     See also `Wikipedia <http://en.wikipedia.org/wiki/Cooley-Tukey_FFT_algorithm>`_.
     """
 
-    # revision 293076305, http://is.gd/1c7PI
+    # revision 293076305
+    # https://en.wikipedia.org/w/index.php?title=Cooley-Tukey_FFT_algorithm&oldid=293076305
+
+    # {{{ parameter processing
+
+    if wrap_intermediate is not None and wrap_intermediate_with_level is not None:
+        raise TypeError("may specify at most one of wrap_intermediate and "
+                "wrap_intermediate_with_level")
+    if wrap_intermediate is not None:
+        from warnings import warn
+        warn("wrap_intermediate is deprecated. Use wrap_intermediate_with_level "
+                "instead. wrap_intermediate will stop working in 2023.",
+                DeprecationWarning, stacklevel=2)
+
+        def wrap_intermediate_with_level(level, x):  # pylint: disable=function-redefined  # noqa: E501
+            return wrap_intermediate(x)
+
+    if wrap_intermediate_with_level is None:
+        def wrap_intermediate_with_level(level, x):
+            return x
 
     from math import pi
-    import numpy
+    if custom_np is None:
+        import numpy as custom_np
 
```

Comment:
To make sure `x` is a numpy array.
```suggestion

x = np.array(x)

```